src/_pytest/fixtures.py

```diff
@@ -1211,21 +1183,56 @@ def __call__(self, function: FixtureFunction) -> FixtureFunction:
                 pytrace=False,
             )
 
-        # Type ignored because https://github.com/python/mypy/issues/2087.
-        function._pytestfixturefunction = self  # type: ignore[attr-defined]
-        return function
+        return fixture_definition
+
+
+class FixtureFunctionDefinition:
```

Comment:
We need to log a followup task for paramspec/return type annotation tracking and storing

Reply:
Added. I just don't fully know what does this mean I added your comment but feel free to edit it with more context.

Reply:
I gave this a try in https://github.com/pytest-dev/pytest/pull/12999, so we don't need to create an issue after all in the end, we can continue on that PR once this one gets merged.